docs/architecture.md

```diff
@@ -0,0 +1,111 @@
+# Architecture
+
+This document describes the high-level architecture of nearcore.
+The focus here is on the implementation of the blockchain protocol, not the protocol itself.
+For reference documentation of the protocol, please refer to [nomicon](https://nomicon.io/)
+
+## Bird's Eye View
+
+![](images/architecture.svg)
+
+Since nearcore is an implementation of NEAR blockchain protocol, its goal is to produce a binary that runs as a blockchain client.
+More specifically, the neard binary can do the following:
+- generate files necessary to start a node
+- start a node with a given folder that contains required data
```

Comment:
It would be helpful to describe "files" in some more detail. "Show me your flowchart and conceal your tables, and I shall continue to be mystified. Show me your tables, and I won't usually need your flowchart; it'll be obvious" and all that. 

In particular, clarifying the relation between genesis file and the rocksdb database would help me, as I am not 100% sure about how genesis works.